### PR TITLE
Kan 6 log in

### PR DESCRIPTION
**Pull request by GeeA02**

### CODE REVIEW

File: server/src/app/features/users/handlers/login.handler.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  async execute({ payload }: LoginCommand) {
+    const { userRepository, tokenRepository, accessTokenService } = this.dependencies;
+    const { username, password } = payload;
+    const user = await userRepository.findOne({ username });
+
+    if (!user || !UserPassword.compare(password, user.salt, user.password)) {
+      throw new Error("Invalid credentials");
+    }
+    //v refreshToken v
```
Comment (Nynu8) [inline server/src/app/features/users/handlers/login.handler.ts]: Do we need those comments here?

File: server/src/app/features/users/handlers/login.handler.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const token = accessTokenService.generateRefreshToken(username);
```
Comment (Nynu8) [inline server/src/app/features/users/handlers/login.handler.ts]: generateRefreshToken takes userId as a parameter, also variable name could be more descriptive

File: server/src/app/features/users/handlers/login.handler.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    //^ refreshToke ^
+    const accessToken = accessTokenService.generateAccessToken(username);
```
Comment (Nynu8) [inline server/src/app/features/users/handlers/login.handler.ts]: same as above

File: server/src/app/features/users/handlers/login.handler.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    let userId = user.id;
```
Comment (Nynu8) [inline server/src/app/features/users/handlers/login.handler.ts]: This should be const

File: server/src/app/features/users/handlers/login.handler.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    const newToken = TokenModel.create({ id: uuid(), userId, token });
+    await tokenRepository.save(newToken);
+
+    return {
```
Comment (Nynu8) [inline server/src/app/features/users/handlers/login.handler.ts:37]: In command you're returning result, so you're actually returning "true" to the user while we need to return the tokens.

**Review by Nynu8 — CHANGES_REQUESTED** [review]

File: server/src/app/features/users/handlers/login.handler.ts (hunk 1 above)
Comment (GeeA02) [reply server/src/app/features/users/handlers/login.handler.ts]: yup

File: server/src/app/features/users/handlers/login.handler.ts (hunk 2 above)
Comment (GeeA02) [reply server/src/app/features/users/handlers/login.handler.ts]: Done

File: server/src/app/features/users/handlers/login.handler.ts (hunk 3 above)
Comment (GeeA02) [reply server/src/app/features/users/handlers/login.handler.ts]: same as above

File: server/src/app/features/users/handlers/login.handler.ts (hunk 4 above)
Comment (GeeA02) [reply server/src/app/features/users/handlers/login.handler.ts]: yup

File: server/src/app/features/users/handlers/login.handler.ts (hunk 5 above)
Comment (GeeA02) [reply server/src/app/features/users/handlers/login.handler.ts:37]: Done

**Review by Nynu8 — APPROVED** [review]

File: server/src/app/features/users/models/token.model.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  public static create(data: Partial<TokenModelProps>): TokenModel {
+    const entity = new TokenModel();
+    Object.assign(entity, data);
+    return entity;
+  }
+
+  @PrimaryGeneratedColumn("uuid")
+  id: string;
+
+  @Column()
```
Comment (Nynu8) [inline server/src/app/features/users/models/token.model.ts:23]: This should be in relation to user

**Review by Nynu8 — CHANGES_REQUESTED** [review]

File: server/src/app/features/users/models/token.model.ts (hunk 6 above)
Comment (GeeA02) [reply server/src/app/features/users/models/token.model.ts:23]: ok

**Pull request merged**